packages/plugin-dates/src/Date.tsx

```diff
@@ -0,0 +1,46 @@
+import { Components, useInputContext } from 'leva/plugin'
+import React, { forwardRef, useCallback } from 'react'
+import DatePicker, { CalendarContainer } from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import { DateCalendarContainerProps, DateInputProps, DateProps } from './date-types'
+import { InputContainer, StyledInput, StyledWrapper } from './StyledDate'
+
+const { Label, Row } = Components
+
+const DateCalendarContainer = ({ children }: DateCalendarContainerProps) => {
+  return (
+    <CalendarContainer>
+      <StyledWrapper>{children}</StyledWrapper>
+    </CalendarContainer>
+  )
+}
+
+const DateInput = forwardRef<HTMLInputElement, Partial<DateInputProps>>(({ value, onClick, onChange }, ref) => {
+  return <StyledInput ref={ref} value={value} onClick={onClick} onChange={onChange} />
+})
+
+export function Date() {
+  const { label, value, onUpdate, settings } = useInputContext<DateProps>()
+
+  const update = useCallback(
```

Comment:
`update` is stable across render, but I'm not sure `update(onChange)` would be.

I think the correct way to do this to prevent unnecessary re-renders in `<DatePicker />` is:

```ts
const onChange = useMemo(() => {
  const update = (fn: (value: Date) => void) => (date: Date) => {
      fn(date)
  }
  return update(onUpdate)
}, [onUpdate])
```

I'm not sure it's needed but I'm pretty positive that `update(onUpdate)` would not be stable, making the use of `useCallback` unnecessary!

Reply:
By the way, can you clarify why we need this at all? Can't we just pass `onUpdate` to the `onChange` prop?

Reply:
@dbismut you're right, currying always gets the better of me, the handler is unnecessary passing `onUpdate` directly works.